File: app/src/main/kotlin/com/youcoded/app/skills/BundledPlugins.kt
// Bundled plugins are marketplace plugins that ship with YouCoded and cannot
// be uninstalled through the UI. On every launch, if a bundled plugin is
// missing, the app reinstalls it silently.
//
// PARITY REQUIRED — keep this list in sync with:
//   youcoded/desktop/src/shared/bundled-plugins.ts
// If you change the list, also update docs/PITFALLS.md.

package com.youcoded.app.skills

object BundledPlugins {
    val IDS = listOf(
        "wecoded-themes-plugin",
        "wecoded-marketplace-publisher",
    )

    const val REASON =
        "Bundled with YouCoded — required for theme customization and publishing."

    fun isBundled(id: String): Boolean = IDS.contains(id)
}

File: desktop/src/shared/bundled-plugins.ts
// Bundled plugins are marketplace plugins that ship with YouCoded and cannot
// be uninstalled through the UI. On every launch, if a bundled plugin is
// missing from ~/.claude/plugins/installed_plugins.json, the app reinstalls
// it silently.
//
// PARITY REQUIRED — keep this list in sync with:
//   youcoded/app/src/main/kotlin/com/youcoded/app/skills/BundledPlugins.kt
// If you change the list, also update docs/PITFALLS.md.

export const BUNDLED_PLUGIN_IDS = [
  'wecoded-themes-plugin',
  'wecoded-marketplace-publisher',
  'youcoded-chatsearch',
  // YouCoded Pages (Phase 1, 2026-09-17): the /page-builder skill behind Make a page and
  // Edit in chat. Bundled so a page can be built without a marketplace visit.
  'wecoded-pages-plugin',
] as const;

export const BUNDLED_REASON =
  'Bundled with YouCoded — required for theme customization, publishing, conversation search, and building pages.';

export function isBundledPlugin(id: string): boolean {
  return (BUNDLED_PLUGIN_IDS as readonly string[]).includes(id);
}
